packages/node-cli/src/commands/account/setupci/AccountSetupCiAction.js

```diff
@@ -5,44 +5,56 @@
 'use strict';
 
 const BaseAction = require('../../base/BaseAction');
-const { authenticateCi } = require('../../../utils/AuthenticationUtils');
-const { DOMAIN: { PRODUCTION: { GENERIC_NETSUITE_DOMAIN } } } = require('../../../ApplicationConstants');
+const { authenticateCi, selectAuthenticationCI } = require('../../../utils/AuthenticationUtils');
+const {
+	DOMAIN: { PRODUCTION: { GENERIC_NETSUITE_DOMAIN } },
+	ACCOUNT_SETUP_CI: { COMMAND: { OPTIONS } },
+} = require('../../../ApplicationConstants');
 const ProjectInfoService = require('../../../services/ProjectInfoService');
 const { validateMachineToMachineAuthIsAllowed } = require('../../../services/ExecutionContextService');
+const AccountSetupCiValidation = require('./AccountSetupCiValidation');
 
-const COMMAND = {
-	OPTIONS: {
-		ACCOUNT: 'account',
-		AUTHID: 'authid',
-		CERTIFCATEID: 'certificateid',
-		PRIVATEKEYPATH: 'privatekeypath',
-		DOMAIN: 'domain'
 
-	},
-	SDK_COMMAND: 'authenticateci',
-};
 
 module.exports = class AccountSetupCiAction extends BaseAction {
-	
+
 	constructor(options) {
 		super(options);
 		this._projectInfoService = new ProjectInfoService(this._projectFolder);
+		this._validator = new AccountSetupCiValidation(this._commandMetadata, this._runInInteractiveMode);
 	}
 
 	preExecute(params) {
 		this._projectInfoService.checkWorkingDirectoryContainsValidProject(this._commandMetadata.name);
-		
-		if (params[COMMAND.OPTIONS.ACCOUNT]) {
-			params[COMMAND.OPTIONS.ACCOUNT] = params[COMMAND.OPTIONS.ACCOUNT].toUpperCase();
+
+		if (params[OPTIONS.ACCOUNT]) {
+			params[OPTIONS.ACCOUNT] = params[OPTIONS.ACCOUNT].toUpperCase();
```

Comment:
I didn't realize about this `.toUpperCase()` change in my first CR pass.
Is this needed? Couldn't it be causing some conflicts in some cases?

We might need to check this  with different account names right? Maybe you have already tested it. 
I'm just leaving the comment so we can check it together.